### PR TITLE
raft: abort leader transferring if the target is demoted

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

Close https://github.com/etcd-io/etcd/issues/11326.

I'm not sure where to add test case for this, in raft_test.go or testdata?

### CODE REVIEW

File: raft/raft.go [hunk 1]
```diff
@@ -1566,8 +1566,8 @@ func (r *raft) switchToConfig(cfg tracker.Config, prs tracker.ProgressMap) pb.Co
 			r.maybeSendAppend(id, false /* sendIfEmpty */)
 		})
 	}
-	// If the the leadTransferee was removed, abort the leadership transfer.
-	if _, tOK := r.prs.Progress[r.leadTransferee]; !tOK && r.leadTransferee != 0 {
+	// If the the leadTransferee was removed or demoted, abort the leadership transfer.
```
Comment (xiang90) [inline raft/raft.go]: can you add a test?

Comment (hicqu) [reply raft/raft.go]: Yes, but I'm not sure the test case should be in raft_test.go or testdata/*.txt.

Comment (xiang90) [reply raft/raft.go]: i think there should be a related test case for the removal one. adding a case there should be fine.

Comment (hicqu) [reply raft/raft.go]: Thank you! I will update the PR soon.

**Comment by xiang90** [comment]
lgtm

**Comment by xiang90** [comment]
@hicqu code needs to be gofmted before merging.

```
@@ -3617,11 +3617,11 @@
 
 	lead.applyConfChange(pb.ConfChangeV2{
 		Changes: []pb.ConfChangeSingle{
-			pb.ConfChangeSingle{
+			{
 				Type:   pb.ConfChangeRemoveNode,
 				NodeID: 3,
 			},
-			pb.ConfChangeSingle{
+			{
 				Type:   pb.ConfChangeAddLearnerNode,
 				NodeID: 3,
 			},
```

**Comment by hicqu** [comment]
PTAL @xiang90 thanks!

**Comment by xiang90** [comment]
lgtm

**Pull request merged**